### PR TITLE
Add a basic mTLS identity tests

### LINKED ISSUE #31: Add a basic MTLS identity test
https://github.com/Kuadrant/authorino/blob/main/docs/user-guides/mtls-authentication.md

### PR DESCRIPTION
**Pull request by averevki**
Closes #31

### CODE REVIEW

File: testsuite/tests/kuadrant/authorino/operator/tls/conftest.py [hunk 1]
```diff
@@ -8,12 +9,19 @@
 
 
 @pytest.fixture(scope="session")
-def certificates(cfssl, authorino_domain, wildcard_domain):
+def cert_attributes_custom() -> Dict[str, str]:
+    """Certificate attributes that are different from the default ones"""
+    return {k: f"{v}-other" for k, v in CFSSLClient.DEFAULT_NAMES[0].items()}
```
Comment (pehala) [inline testsuite/tests/kuadrant/authorino/operator/tls/conftest.py:14]: I would rather copy this or create a minimalistic version of this dict, I do not like idea of depending on the defaults

File: testsuite/tests/kuadrant/authorino/operator/tls/mtls/test_mtls_identity.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+@pytest.fixture(scope="module", autouse=True)
+def authorization(authorization, blame, selector_params):
+    """Create AuthConfig with mtls identity and pattern matching rule"""
+    authorization.remove_all_identities()
+
+    authorization.add_mtls_identity(blame("mtls"), selector_params[0], selector_params[1])
+    authorization.add_json_rule(name=blame("redhat"),
+                                selector="auth.identity.Organization",
+                                operator="incl",
+                                value=CFSSLClient.DEFAULT_NAMES[0]["O"])
```
Comment (pehala) [inline testsuite/tests/kuadrant/authorino/operator/tls/mtls/test_mtls_identity.py:17]: Same here

**Review by pehala — APPROVED** [review]

**Review by pehala — DISMISSED** [review]

File: testsuite/tests/kuadrant/authorino/operator/tls/conftest.py [hunk 3]
```diff
@@ -8,12 +9,29 @@
...
+    """Certificate attributes"""
+    return dict(O="Organization Test",
+                OU="Unit Test",
+                L="Location Test",
+                ST="State Test",
+                C="Country Test")
+
+
+@pytest.fixture(scope="session")
+def cert_attributes_custom(cert_attributes) -> Dict[str, str]:
```
Comment (guicassolato) [inline testsuite/tests/kuadrant/authorino/operator/tls/conftest.py:22]: It looks like this method should be called `cert_attributes_other`, as you cannot really customise the values. The values will always be whatever you supplied in the provided cert, suffixed by `-other`.

```suggestion
def cert_attributes_other(cert_attributes) -> Dict[str, str]:
```

File: testsuite/tests/kuadrant/authorino/operator/tls/mtls/test_mtls_identity.py [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+
+from testsuite.objects import Rule
+
+
+@pytest.fixture(scope="module", autouse=True)
+def authorization(authorization, blame, selector_params, cert_attributes):
+    """Create AuthConfig with mtls identity and pattern matching rule"""
+    authorization.remove_all_identities()
+
+    authorization.add_mtls_identity(blame("mtls"), selector_params[0], selector_params[1])
```
Comment (guicassolato) [inline testsuite/tests/kuadrant/authorino/operator/tls/mtls/test_mtls_identity.py:13]:
```suggestion
    authorization.add_mtls_identity(blame("mtls"), *selector_params)
```

Too idiomatic?

**Review by guicassolato — COMMENTED** [review]
Any use case for a X.509 cert that is accepted in Envoy, but rejected by Authorino with a 401 instead of a 403?

File: testsuite/tests/kuadrant/authorino/operator/tls/conftest.py (hunk 3 above)
Comment (pehala) [reply testsuite/tests/kuadrant/authorino/operator/tls/conftest.py:22]: You can actually customize them, in `pytest` you can override fixtures in the individual test files so it is expected that some other test case will change this fixture and supply different attributes. 

From a naming perspective tho, I have no preference either way

**Comment by averevki** [comment]
> Any use case for a X.509 cert that is accepted in Envoy, but rejected by Authorino with a 401 instead of a 403?

Will be covered in another PR for the more advanced tests (#137). Thank you for the idea!

**Review by guicassolato — APPROVED** [review]

**Review by pehala — APPROVED** [review]

**Pull request merged**